nixio/cmd/nixexplore.py

```diff
@@ -0,0 +1,623 @@
+        print(dim_ticks_conv_func1(first_dim_ticks[i]) + "    " + "   ".join(map(data_conv_func, data[i, :])))
+        if i % 500 == 0:
+            progress(i, data.shape[0], status='Dumping ...')
+    print(end)
+
+
+def dump_threed(data, dimensions, label, unit, format="%.6f", end="\n\n"):
+    assert(len(data.shape) == 3 and len(dimensions) == 3)
+    ticks = get_ticks(dimensions[2], data.shape[2])
+    dim_label, dim_unit = get_dim_label_and_unit(dimensions[2])
+
+    for i in range(data.shape[2]):
+        print("# data[:, :, %i]: %s" % (i, dim_label + "%s%s" % (ticks[i], dim_unit) if dim_unit else ""))
+        dump_twod(data[:, :, i], [dimensions[0], dimensions[1]], label, unit, format, end="\n")
+
+    print(end)
+
+
+def dump_data_array(array, filename):
+    print("# File: %s\n# entity: %s\n# type: %s\n# id: %s" % (filename, array.name, array.type, array.id))
+    print("# created at: %s\n# last edited at: %s\n" %
+          (str(dt.datetime.fromtimestamp(array.created_at)),
+           str(dt.datetime.fromtimestamp(array.updated_at))))
+    dims = len(array.shape)
+    data = array[:]
+    if dims == 1:
+        dump_oned(data, array.dimensions[0], array.label, array.unit)
+    elif dims == 2:
+        dump_twod(data, array.dimensions, array.label, array.unit)
+    elif dims == 3:
+        dump_threed(data, array.dimensions, array.label, array.unit)
+    else:
+        print("Sorry, cannot dump data with more than 3 dimensions!")
+
+
+def data_dump(filename, arguments):
+    nix_file = open_nix_file(filename)
+    entities = find_data_entity(nix_file, arguments)
+    for e in entities:
+        if isinstance(e, nix.pycore.data_array.DataArray):
+            dump_data_array(e, filename)
+
+    nix_file.close()
+
+
+def data_plotter(filename, arguments):
+    nix_file = open_nix_file(filename)
+    entities = find_data_entity(nix_file, arguments)
+    for e in entities:
+        if isinstance(e, nix.pycore.data_array.DataArray):
+            plotter = nw.plotter.suggested_plotter(e)
+            if plotter:
+                plotter.plot()
+                plt.show()
+            else:
+                print("Could not find a suitable plotter for the DataArray: %s" % str(e))
+        else:
+            print("Sorry, so far I can only try to plot DataArrays.")
+
+    nix_file.close()
+
+
+def disp_data(filename, arguments):
+    nix_file = open_nix_file(filename)
+    entities = find_data_entity(nix_file, arguments)
+    print("# File: %s" % filename)
+    for e in entities:
+        print("# entity: %s\n# type: %s\n# id: %s" % (e.name, e.type, e.id))
+        print("# created at: %s\n# last edited at: %s\n" %
+              (str(dt.datetime.fromtimestamp(e.created_at)),
+               str(dt.datetime.fromtimestamp(e.updated_at))))
+        print(e)
+        print("\n")
+    nix_file.close()
+
+
+def data_worker(arguments):
+    files = assemble_files(arguments)
+    func = disp_data
+    for nf in files:
+        func(nf, arguments)
```

Comment:
```suggestion
    for nf in files:
        disp_data(nf, arguments)
```

This function name assignment seems unnecessary.